lib/pbench/test/unit/server/test_upload.py

```diff
@@ -227,16 +227,16 @@ def test_bad_metadata_upload(self, client, server_config, pbench_drb_token):
                     "Content-MD5": "ANYMD5",
                 },
                 query_string={
-                    "metadata": "global.xyz#A@b=z:y,foobar.badpath:data,server.deletion:3000-12-25T23:59:59+00:00"
+                    "metadata": "global.xyz#A@b=z:y,foobar.badpath:data,server.deletion:'3000-12-25T23:59:59+00:00'"
                 },
             )
         assert response.status_code == HTTPStatus.BAD_REQUEST
         json = response.json
         assert "errors" in json and "message" in json
         assert json["message"] == "at least one specified metadata key is invalid"
         assert json["errors"] == [
-            "Key global.xyz#a@b=z is invalid or isn't settable",
-            "Key foobar.badpath is invalid or isn't settable",
+            "Metadata key 'global.xyz#A@b=z' is not supported",
+            "Metadata key 'foobar.badpath' is not supported",
```

Comment:
Why are these two keys now "not supported"?  The first one because it has funky characters in it?  The second one because the top-level namespace doesn't exist (and isn't allowed to be created)?

Are we satisfied with these error messages?

Reply:
I decided I didn't really like the "invalid or isn't settable" and generalized the wording; but it's the same thing. I can't really break up the two cases, because key validation works off a preconfigured list that differs for read vs write. A key that "isn't settable" *is* "invalid" (or "not supported") for write.